core/divelist.c

```diff
@@ -1424,8 +1424,7 @@ void clear_dive_file_data()
 	clear_dive(&displayed_dive);
 	clear_dive_site(&displayed_dive_site);
 
-	free((void *)existing_filename);
-	existing_filename = NULL;
+	set_current_file_none();
```

Comment:
clear_current_file() ? or set_current_file(NULL)? This seems like an odd function